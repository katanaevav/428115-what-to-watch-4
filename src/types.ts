export interface Movie {
  id: number,
  title: string,
  smallPoster: string,
  genre: string,
  year: number,
  runTime: number,
  bigPoster: string,
  cover: string,
  ratingScore: number,
  ratingCount: number,
  preview: string,
  video: string,
  directors: Array<string>,
  starrings: Array<string>,
  descriptions: Array<string>,
  backgroundColor: string,
  isFavorite: boolean,
}

export interface Comment {
  id: number,
  text: string,
  author: string,
  date: number,
  mark: number,
}
