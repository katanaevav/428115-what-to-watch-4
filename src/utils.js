export const emptyFunction = () => {
  // do nothing
};
