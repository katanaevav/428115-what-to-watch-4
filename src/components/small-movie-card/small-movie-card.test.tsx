import * as React from "react";
import * as renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card";
import {Router} from "react-router-dom";
import history from "../../history";
import {Movie} from "../../types";
import {emptyFunction} from "../../utils";

const movie: Movie = {
  id: 0,
  title: `Bohemian rhapsody`,
  smallPoster: `img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2018,
  bigPoster: `img/bohemian-rhapsody.jpg`,
  cover: `img/bohemian-rhapsody.jpg`,
  ratingScore: 7.9,
  runTime: 170,
  ratingCount: 284651,
  isFavorite: true,
  backgroundColor: `#AABBCC`,
  video: ``,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  directors: [`Брайан Сингер`],
  starrings: [`Рами Малек, Люси Бойнтон, Гвилим Ли, Бен Харди, Джозеф Маццелло, Эйдан Гиллен, Аллен Лич, Том Холландер, Майк Майерс, Аарон МакКаскер и другие`],
  descriptions: [`Чествование группы Queen, их музыки и их выдающегося вокалиста Фредди Меркьюри, который бросил вызов стереотипам и победил условности, чтобы стать одним из самых любимых артистов на планете. Фильм прослеживает головокружительный путь группы к успеху благодаря их культовым песням и революционному звуку, практически распад коллектива, поскольку образ жизни Меркьюри выходит из-под контроля, и их триумфальное воссоединение накануне концерта Live Aid, ставшим одним из величайших выступлений в истории рок-музыки.`],
};

it(`Render small movie card`, () => {
  const tree = renderer.create(
      <Router history = {history}>
        <SmallMovieCard
          key = {movie.id}
          movieId = {movie.id}
          movieTitle = {movie.title}
          movieSmallPoster = {movie.smallPoster}
          preview = {movie.preview}
          onMovieMouseOver = {emptyFunction}
          onMovieMouseOut = {emptyFunction}
          renderPlayer = {() => null}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
