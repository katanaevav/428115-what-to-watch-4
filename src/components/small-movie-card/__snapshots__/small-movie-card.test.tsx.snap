// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render small movie card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-key={0}
>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/0"
      onClick={[Function]}
    >
      Bohemian rhapsody
    </a>
  </h3>
</article>
`;
